src/MarginTrading.SettingsService/Controllers/TradingInstrumentsController.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MarginTrading.SettingsService.AzureRepositories.Entities;
+using MarginTrading.SettingsService.Client;
+using MarginTrading.SettingsService.Client.TradingConditions;
+using MarginTrading.SettingsService.Core.Domain;
+using MarginTrading.SettingsService.Core.Services;
+using MarginTrading.SettingsService.Core.Settings;
+using MarginTrading.SettingsService.StorageInterfaces.Repositories;
+using Microsoft.AspNetCore.Mvc;
+
+namespace MarginTrading.SettingsService.Controllers
+{
+    /// <summary>
+    /// Trading instruments management
+    /// </summary>
+    [Route("api/tradingInstruments")]
+    public class TradingInstrumentsController : Controller, ITradingInstrumentsApi
+    {
+        private readonly ITradingInstrumentsRepository _tradingInstrumentsRepository;
+        private readonly ITradingService _tradingService;
+        private readonly IConvertService _convertService;
+        private readonly IEventSender _eventSender;
+        private readonly DefaultTradingInstrumentSettings _defaultTradingInstrumentSettings;
+        
+        public TradingInstrumentsController(
+            ITradingInstrumentsRepository tradingInstrumentsRepository,
+            ITradingService tradingService,
+            IConvertService convertService,
+            IEventSender eventSender,
+            DefaultTradingInstrumentSettings defaultTradingInstrumentSettings)
+        {
+            _tradingInstrumentsRepository = tradingInstrumentsRepository;
+            _tradingService = tradingService;
+            _convertService = convertService;
+            _eventSender = eventSender;
+            _defaultTradingInstrumentSettings = defaultTradingInstrumentSettings;
+        }
+        
+        /// <summary>
+        /// Get the list of trading instruments
+        /// </summary>
+        /// <param name="tradingConditionId"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("")]
+        public async Task<List<TradingInstrumentContract>> List([FromQuery] string tradingConditionId)
+        {
+            var data = string.IsNullOrWhiteSpace(tradingConditionId)
+                ? await _tradingInstrumentsRepository.GetAsync()
+                : await _tradingInstrumentsRepository.GetAsync(x => x.TradingConditionId == tradingConditionId);

```

Comment:
Look at proposal in TradingInstrumentEntity and I think all get logic should be changed